contracts/src/models/resources.cairo

```diff
@@ -50,6 +50,10 @@ impl ResourceImpl of ResourceTrait {
 
         // take the balance from the production balance and the physical 
         self.balance += production_balance.balance().into();
+
+        // todo@credence we also need to deduct from balance if resource is
```

Comment:
this should happen in the production trait - there already is a function to determine this